common/src/main/java/com/zegelin/prometheus/netty/Server.java

```diff
@@ -62,27 +70,44 @@ public static void start(final List<InetSocketAddress> listenAddresses, final Ha
                 .channel(NioServerSocketChannel.class)
                 .childHandler(new ChildInitializer(harvester, helpExposition));
 
-        final List<Channel> channels;
         {
             final ImmutableList.Builder<Channel> builder = ImmutableList.builder();
 
             for (final InetSocketAddress listenAddress : listenAddresses) {
-                final Channel channel = bootstrap.bind(listenAddress).sync().channel();
-
-                builder.add(channel);
+                builder.add(bootstrap.bind(listenAddress).sync().channel());
             }
 
-            channels = builder.build();
+            Server.channels = builder.build();
         }
 
         if (logger.isInfoEnabled()) {
-            logger.info("cassandra-exporter started. Listening on {}", Joiner.on(", ").join(
+            logger.info("Cassandra Exporter Agent Server has started. Listening on {}", Joiner.on(", ").join(
                     listenAddresses.stream()
                             .map(a -> String.format("http://%s:%d", a.getHostString(), a.getPort()))
                             .iterator()
             ));
         }
+    }
+
+    public static void stop() {
+
+        if (Server.channels != null) {
 
-        // TODO: maybe return a future that the caller can sync on, to wait for the channels to shutdown?
+            for (Channel ch : channels) {
```

Comment:
Minor nit: I usually declare all local variables as `final` unless they get reassigned in-scope.